scripts/backup_with_rotation.sh

```
@@ -0,0 +1,43 @@
+	zip -r  "${backup_dir}.zip" "${source_dir}" >/dev/null
+
+	if [[ $? -eq 0 ]];
+	then
+		echo "Backup created successfully: ${backup_dir}.zip"
+	else
+		echo "Backup failed to create."
+	fi
+}
+
+function backup_rotation {
+	backups=($(ls -t "${source_dir}/backup/backup_"*.zip 2>/dev/null))
```

Shellcheck

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)